src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -1209,12 +1213,33 @@ ep_buffer_manager_write_all_buffers_to_file_v4 (
 					// miscategorize it, but that seems unlikely.
 					uint32_t last_read_delta = last_read_sequence_number - thread_sequence_number;
 					if (0 < last_read_delta && last_read_delta < 0x80000000) {
-						ep_rt_thread_sequence_number_map_add_or_replace (ep_sequence_point_get_thread_sequence_numbers_ref (sequence_point), session_state, last_read_sequence_number);
```

Comment:
Any reason why add_or_replace couldn't be used? In theory that method could be a little more effective when hitting the replace scenario (depending on internal implementation).

Reply:
SHash in Coreclr is implemented so it can't have the remove trait _and_ the AddOrReplace trait on it. I didn't look into why, but I see this as more evidence we may want to eventually have a shared set of containers. It would remove ambiguity and runtime differences like this.

Reply:
OK, since you needed additional remove scenario you hit this with both behaviours on the hash map and needed to adjust. For Mono's hash map you could have both, feels like extending SHash map to have both could make sense as well.

Reply:
Looks like limitation is in `SHash<TRAITS>::AddOrReplace`:

`static_assert(!TRAITS::s_supports_remove, "SHash::AddOrReplace is not implemented for SHash with support for remove operations.");`

Reply:
The add_or_replace function on hash map was done when porting this use case over to C library, if we nuke the user of add_or_replace, then I think it will be good to remove that function from hash map implementation altogether.